library/std/src/sys/sync/rwlock/futex.rs

```diff
@@ -101,11 +117,12 @@ impl RwLock {
 
     #[cold]
     fn read_contended(&self) {
+        let mut has_slept = false;
         let mut state = self.spin_read();
 
         loop {
             // If we can lock it, lock it.
-            if is_read_lockable(state) {
+            if has_slept && is_read_lockable_after_wakeup(state) || is_read_lockable(state) {
```

Comment:
Could you update the above comment to mention this check?

Optional nit to make the precedence more readable

```suggestion
            if (has_slept && is_read_lockable_after_wakeup(state)) || is_read_lockable(state) {
```